src/main/java/nwoolcan/model/brewery/production/batch/step/BasicStepFactory.java

```diff
@@ -0,0 +1,26 @@
+package nwoolcan.model.brewery.production.batch.step;
+
+import nwoolcan.utils.Result;
+
+import java.util.Date;
+
+/**
+ * Step factory implementation for creating basic steps.
+ */
+public final class BasicStepFactory implements StepFactory {
+
+    private static final String CANNOT_FIND_STEP_IMPLEMENTATION_MESSAGE = " does not have a configured implementation.";
+
+    @Override
+    public Result<Step> create(final StepType stepType, final Date startDate) {
+        if (stepType instanceof StepTypeEnum) {
```

Comment:
```suggestion
        return Result.ofEmpty()
                            .require(stepType instanceof StepTypeEnum, 
                               new IllegalArgumentException(stepType + CANNOT_FIND_STEP_IMPLEMENTATION_MESSAGE)
                            .map(new BasicStep(stepType, startDate)
```

Reply:
just because i like functional style. If you don't you can leave it as you prefer